src/webhooks/registry.ts

```diff
@@ -132,6 +124,10 @@ function buildQuery(
   deliveryMethod: DeliveryMethod = DeliveryMethod.Http,
   webhookId?: string,
 ): string {
+  if (deliveryMethod === DeliveryMethod.EventBridge && !versionSupportsEndpointField()) {
+    // No endpoint fallback exists for EventBridge, so throw early
+    throw new ShopifyErrors.UnsupportedClientType(`EventBridge webhooks are not supported in API version "${Context.API_VERSION}".`);
+  }
```

Comment:
We may want to run this in `buildCheckQuery` as well, since that will be called before this. We could actually just run this as a gateway check in `register` to fail as early as we can.

Reply:
I wasn't quite sure what to do here. I originally moved the check in `register` as you suggested, but realised that the `buildQuery` method is exported, so could potentially be consumed and run elsewhere without error being thrown (e.g. in tests), I guess the same is true of `buildCheckQuery`. What would you suggest? Do you think it just being in `register` is sufficient?

Reply:
Hm, good catch - IMHO if the methods are exported we might as well go for the safer choice and just run the check on all exported methods to make sure we're covering all cases, it's cheaper to run it multiple times than it is to make a network request and wait for it to fail.

Reply:
fb6e23d32012d830b935a1ca961f901b96ae43a5 adds a common `validateDeliveryMethod` function which is executed on `register` and `buildQuery` as both those function have access to the `deliveryMethod`.

I suppose `buildCheckQuery` is perhaps fine without the check as it doesn't actually care which delivery method you are using and only accepts a topic as a parameter? For instance there may be a use case where you would want to query for an already registered webhook without knowing what type of delivery method that webhook is actually utilising.